_posts/pl/2024-12-16-czy-wiesz-ze-w-angular-17-zostala-wprowadzona-alternatywa-dla-ngfor.md

```diff
@@ -0,0 +1,47 @@
+---
+layout:    post
+title:     Czy wiesz, że w Angular 17 została wprowadzona alternatywa dla *ngFor?
+description: ""
+date:      2024-12-16T08:00:00+01:00
+published: true
+didyouknow: true
+lang: pl
+author: dmejer
+image: /assets/img/posts/2024-12-16-czy-wiesz-ze-w-angular-17-zostala-wprowadzona-alternatywa-dla-ngfor/thumbnail.webp
+tags:
+- angular
+---
+W Angular 17 pojawiło się *built-it control flow*, które zostało ustabilizowane w wersji 18. Są to zamienniki dyrektyw `*ngIf` (opisaliśmy to [tutaj]({% post_url pl/2024-11-04-czy-wiesz-ze-od-angular-17-mozna-uzywac-if-zamiast-ngif %})), `*ngFor`, `*ngSwitch`.
+
+`@for` jest odpowiednikiem `*ngFor`. Przykładowe wykorzystanie:
+```
+// *ngFor
+<li *ngFor="let item of items; trackBy: trackByFn">{{ item.name }}</li>
+ 
+// @for
+@for (item of items; track item.id) {
+  <li>{{ item.name }}</li>
+}
+```
+
+`track`, tak jak wcześniej, łączy dane z konkretnym węzłem w DOM. Powoduje to wzrost wydajności przy operacjach na kolekcji. Ustawienie `trackBy` było opcjonalne w `*ngFor`, natomiast `track` w `@for` jest obowiązkowy.
```

Comment:
kosmetyczna sugestia dot. szyku zdania:
W `*ngFor` ustawienie `trackBy` było opcjonalne, natomiast w `@for` `track` jest obowiązkowy.